### PR TITLE
daemon: check for dependencies versions at runtime

### PR DESCRIPTION
**Pull request by pkalever**
### What does this PR achieve? Why do we need it?

defend on minimum targetcli, tcmu-runner versions.

### Does this PR fix issues?

Fixes: #76
Fixes: #118

### Notes for the reviewer

I hope, we are good with,

Requesites:
targetcli   >= 2.1.fb46
tcmu-runner >= 1.1.3

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

### CODE REVIEW

**Comment by pkalever** [comment]
@amarts @lxbsz please review. Thanks!

File: daemon/gluster-blockd.c [hunk 1]
```diff
@@ -366,22 +352,114 @@ gbMinKernelVersionCheck(void)
...
+  return;
+
+ out:
+  GB_FREE(out);
+  LOG("mgmt", GB_LOG_ERROR, "%s",
+      "Please install the recommended dependency version and try again.");
+  exit(EXIT_FAILURE);
+}
+
+
```
Comment (amarts) [inline daemon/gluster-blockd.c]: While all the above looks fine, it makes me sad that we need to write so much code in C to do it :-/

No other option though.

Comment (pkalever) [reply daemon/gluster-blockd.c]: Indeed :-(

**Comment by pkalever** [comment]
Thanks @lxbsz @amarts for the review. Updated the patches with tags. Merging now.

**Pull request merged**